### ISSUE TITLE
[BUG] Search does not find with absolute title

### ISSUE THREAD

**Issue by neo42JBR** [mealie-recipes/mealie#4392]
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [X] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

When using recipe search with a long title the recipes will not be matched.

I'd expect a 100% match on title to always result in the matching recipe to be the first result.

I was looking for

> Vegan Carnitas! Hot Pulled Mushroom Tacos mit Limetten-Mayonnaise und Avocado
and the first result was
> Steinpilz-Risotto mit Kürbis 

with the search recipe not even appearing on the list

### Steps to Reproduce

1) Choose a recipe with a long name
2) Copy the recipe title
3) Use search

### Please provide relevant logs

I am not aware of which error logs would be relevant

### Mealie Version

v1.12.0 

### Deployment

Unraid

**Comment by michael-genson** [comment]
I can't reproduce this. Which backend are you using, sqlite or postgres? Can you share a few titles that don't work?

**Comment by ProjectOrangeJuice** [comment]
I have an issue with having `-` in the search, which looks like it might be the issue here instead of long titles?

For me, the tile is "Chicken katsu with extra-veg sauce" does not work. If i edit the title and remove the `-` it works fine, `extra - veg` still does not.

I'm using sqlite

Here looks to be a similar issue https://github.com/mealie-recipes/mealie/issues/4208

**Comment by michael-genson** [comment]
This is probably an issue with sqlite, as it doesn't support searching as well as postgres does. While differences in functionality between the two are minimal in Mealie, search is one of the main differences

**Comment by neo42JBR** [comment]
Yes I was using SQLite because it came packaged with the container. Guess I will just live with it

**Comment by ctdenney** [comment]
I've replicated this on the demo site. Any exact-match search that contains a "-" will fail. I've also tested a couple other special characters (like !, :, and |) and found that it doesn't fail in those cases. So the problem appears to specifically be with "-"

**Comment by neo42JBR** [comment]
I migrated to a PSQL DB and the issue disappeared there

**Comment by henry1113nz** [comment]
This looks like it was fixed by #7603, which landed in May 2026.

The cause was a mismatch between the two halves of normalization. `SearchFilter` in
`query_search.py` has always mapped punctuation to spaces before tokenizing, but
`SqlAlchemyBase.normalize`, which produces the stored `name_normalized`, did not:

    return unidecode(val).lower().strip()[:255]

So "Chicken katsu with extra-veg sauce" was stored with the hyphen intact, while a search for
the same string was tokenized into `extra` and `veg` - neither of which matched the stored
`extra-veg`.

That also explains the two things that seemed odd in this thread. @ctdenney found that `-`
failed while `!`, `:` and `|` did not: hyphens tend to sit *inside* a word, so splitting there
breaks the token, whereas the others usually sit at a boundary where splitting is harmless. And
@neo42JBR's move to Postgres fixed it because `SearchFilter` routes Postgres to fuzzy search and
SQLite to the tokenized path, so only SQLite ever saw the mismatch. The SQLite/Postgres
difference was real, but it was exposing the normalization gap rather than being the cause.

#7603 made `normalize` apply the same punctuation mapping, and shipped a migration to
re-normalize existing rows.

I checked the current behaviour against `mealie-next` on SQLite, using the titles from this
thread:

| search | result |
| --- | --- |
| `Chicken katsu with extra-veg sauce` | 1 hit, correct recipe first |
| `extra-veg` | 1 hit, correct recipe first |
| `Steinpilz-Risotto` | 1 hit, correct recipe first |

Could this be closed?
